konvoy: Update docs for konvoy release v1.5.x-dk (#2836)
mesosphere/dcos-docs-site: pages/ksphere/konvoy/latest/install/install-secured/index.md

---
layout: layout.pug
navigationTitle: Install on secured machines
title: Install on secured machines
menuWeight: 37
excerpt: Install on secured machines
enterprise: false
---

<!-- markdownlint-disable MD004 MD007 MD025 MD030 -->

> The topics in this section guide you through the basic steps to prepare your environment and install Konvoy in an air-gapped environment.

Reviewer: ```suggestion
This section describes how to prepare your environment and install Konvoy in an air-gapped environment.
```

> **IMPORTANT** air-gapped installation is still in Beta and the process may change in the future.

Reviewer: ```suggestion
<p class="message--important"><strong>IMPORTANT: </strong>Air-gapped installation is still in Beta and the process may change in the future.</p>

```

# Before you begin

> Before installing, verify that your environment meets the following basic requirements:

Reviewer: ```suggestion
Before installing, ensure that your environment has the following basic requirements:
```

* [Docker Desktop][install_docker] version 18.09.2 or later

  You must have Docker Desktop installed on the host where the Konvoy command line interface (CLI) will run.
  For example, if you are installing Konvoy on your laptop, be sure the laptop has a supported version of Docker Desktop.

* [kubectl][install_kubectl] v1.16.8 or later

  To enable interaction with the running cluster, you must have `kubectl` installed on the host where the Konvoy command line interface (CLI) will run.

> * The `konvoy_air_gapped.tar.bz2` that will contain the required artifacts to perform an air-gapped installation.

Reviewer: ```suggestion
* The `konvoy_air_gapped.tar.bz2` that contains the required artifacts to perform an air-gapped installation.
```

## Control plane nodes

* You should have at least three control plane nodes.

* Each control plane node should have at least:
  * 4 cores
  * 16 GiB memory
  * 80 GiB of free space in the root partition, and the root partition must be less than 85% full.

## Worker nodes

* You should have at least four worker nodes.

>   The specific number of worker nodes required for your environment can vary depending on the cluster workload and size of the nodes.

Reviewer: ```suggestion
  The specific number of worker nodes required for your environment varies depending on the cluster workload and size of the nodes.
```

* Each worker node should have at least:
  * 8 cores
  * 32 GiB memory
  * 80 GiB of free space in the root partition and the root partition must be less than 85% full.

* If you plan to use **local volume provisioning** to provide [persistent volumes][persistent_volume] for the workloads, you must mount at least three volumes to `/mnt/disks/` mount point on each node.
  Each volume must have **at least** 55 GiB of capacity if the default addon configurations are used.

## Operating system and services for all nodes

For all hosts that are part of the cluster -- except the **deploy host** -- you should verify the following configuration requirements:

* Firewalld is disabled.
* Containerd is uninstalled.
* Docker-ce is uninstalled.
* Swap is disabled.

## Installation

On highly secured clusters you may need to modify the `cluster.yaml` file with additional options.

> See below for possible changes that may be applied in your cluster.

Reviewer: ```suggestion
See the sample file below for possible changes that may be applied in your cluster.
```
